File: exercise/1-two-sum/Mister-Hope.py
# Runtime: 476 ms, faster than 32.13% of Python3 online submissions for Two Sum.
#
# Memory Usage: 27.9 MB, less than 17.63% of Python3 online submissions for Two Sum.

class Solution:
  def twoSum(self, nums: List[int], target: int) -> List[int]:
    prevMap = {}
    
    for i, v in enumerate(nums):
      j = target - v
      if j in prevMap:
        return [prevMap[j], i]
      prevMap[v] = i
    return

File: exercise/1-two-sum/Mister-Hope.js
/*
 * Runtime: 112 ms, faster than 48.98% of JavaScript online submissions for Two Sum.
 *
 * Memory Usage: 48.3 MB, less than 5.73% of JavaScript online submissions for Two Sum.
 */

const twoSum = (nums, target) => {
  const arr = [];

  for (let i = 0; i < nums.length; i++) {
    let j = arr[target - nums[i]];
    if (j !== undefined) return [j, i];
    arr[nums[i]] = i;
  }

  throw new Error("No solution");
};
